provider/resource_edge_deployment.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceEdgeDeployment() *schema.Resource {
+	return &schema.Resource{
+		Create:   createOrUpdateEdgeDeployment,
+		Read:     readEdgeDeployment,
+		Update:   createOrUpdateEdgeDeployment,
+		Delete:   deleteEdgeDeployment,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			"site_short_name": {
+				Type:        schema.TypeString,
+				Description: "Site short name",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func createOrUpdateEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
+	sc := pm.Client
```

Comment:
Why copy the client?

Reply:
This is another thing I got from existing resources. I'll change it to `pm.Client.CreateOrUpdateEdgeDeployment` instead.